DigitalLearningSolutions.Data/DataServices/UserDataService.cs

```diff
@@ -294,5 +297,61 @@ public void ApproveDelegateUsers(params int[] ids)
                 new { ids }
             );
         }
+
+        public void RemoveDelegateUser(int delegateId)
+        {
+            connection.Execute(
+                @"
+                    DELETE FROM AssessAttempts
+	                    WHERE CandidateID = @delegateId
+
+                    DELETE FROM NotificationUsers
+                        WHERE CandidateID = @delegateId
+
+                    DELETE FROM CandidateAssessments
+                        WHERE CandidateID = @delegateId
+
+                    DELETE FROM FilteredLearningActivity
+                        WHERE CandidateID = @delegateId
+
+                    DELETE FROM GroupDelegates
+                        WHERE DelegateID = @delegateId
+
+                    DELETE FROM kbVideoTrack
+                        WHERE CandidateID = @delegateId
+
+                    DELETE FROM ReflectiveAccounts
+                        WHERE CandidateID = @delegateId
+
+                    DELETE FROM SelfAssessmentComments
+                        WHERE CandidateID = @delegateId
+
+                    DELETE FROM SelfAssessmentResults
+                        WHERE CandidateID = @delegateId
+
+                    DELETE FROM Sessions
+                        WHERE CandidateID = @delegateId
+
+                    DELETE FROM aspProgressLearningLogItems
+	                    WHERE aspProgressID IN (SELECT aspProgressID FROM aspProgress
+                            WHERE ProgressID IN (SELECT ProgressID FROM Progress WHERE CandidateID = @delegateId))
+
+                    DELETE FROM aspProgress
+	                    WHERE ProgressID IN (SELECT ProgressID FROM Progress WHERE CandidateID = @delegateId)
+
+                    DELETE FROM ProgressLearningLogItems
+	                    WHERE ProgressID IN (SELECT ProgressID FROM Progress WHERE CandidateID = @delegateId)
+
+                    DELETE FROM ProgressContributors
+	                    WHERE CandidateID = @delegateId
+
+                    DELETE FROM Progress
+                        WHERE CandidateID = @delegateId
+
+                    DELETE FROM Candidates
+                        WHERE CandidateID = @delegateId",

```

Comment:
Do we want to wrap this in a transaction just in case something somehow goes wrong. (I don't really see how it would)

Reply:
No real reason not to I think, and I always favour paranoia around deletes.